src/Illuminate/Auth/Access/Gate.php

```diff
@@ -356,14 +375,16 @@ protected function callAfterCallbacks($user, $ability, array $arguments, $result
         $arguments = array_merge([$user, $ability, $result], [$arguments]);
 
         foreach ($this->afterCallbacks as $after) {
-            $after(...$arguments);
+            if ($user || $after['options']['guestable']) {
+                $after['callback'](...$arguments);
+            }
```

Comment:
This doesn't make any sense. Why are you executing the after callback if there's a user or if it's guestable?